File: examples/spacetrack/filter_comparison.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Demonstrates comparison operators in SpaceTrack queries.
"""

import os

import brahe as bh

# Get credentials from environment
# Note: For production use, most users should use the default constructor:
#   client = bh.SpaceTrackClient(user, password)
# These examples use the test server for automated testing.
user = os.environ.get("TEST_SPACETRACK_USER", "")
password = os.environ.get("TEST_SPACETRACK_PASS", "")

if not user or not password:
    print("Set TEST_SPACETRACK_USER and TEST_SPACETRACK_PASS environment variables")
else:
    client = bh.SpaceTrackClient(
        user, password, base_url="https://for-testing-only.space-track.org/"
    )

    # Greater than: epochs after a date
    recent = client.gp(
        epoch=">2024-01-01",
        object_type="PAYLOAD",
        limit=5,
    )
    print(f"Found {len(recent)} records with epoch > 2024-01-01")

    # Less than: objects launched before a date
    old_sats = client.satcat(
        launch="<1970-01-01",
        current="Y",  # Still in orbit
        limit=5,
    )
    print(f"Found {len(old_sats)} satellites launched before 1970 still in orbit")

    # Not equal: exclude debris
    non_debris = client.gp(
        object_type="<>DEBRIS",
        limit=5,
    )
    print(f"Found {len(non_debris)} non-debris objects")
    # Found 5 records with epoch > 2024-01-01
    # Found 5 satellites launched before 1970 still in orbit
    # Found 5 non-debris objects

print("Example completed successfully!")

File: examples/spacetrack/filter_comparison.rs
//! Demonstrates comparison operators in SpaceTrack queries.
//!
//! FLAGS = ["CI-ONLY"]

use brahe::spacetrack::{BlockingSpaceTrackClient, TEST_BASE_URL};

fn main() {
    // Get credentials from environment
    // Note: For production use, most users should use the default constructor:
    //   BlockingSpaceTrackClient::new("user", "password")
    // These examples use the test server for automated testing.
    let user = std::env::var("TEST_SPACETRACK_USER").unwrap_or_default();
    let password = std::env::var("TEST_SPACETRACK_PASS").unwrap_or_default();

    if user.is_empty() || password.is_empty() {
        println!("Set TEST_SPACETRACK_USER and TEST_SPACETRACK_PASS environment variables");
    } else {
        let client = BlockingSpaceTrackClient::with_base_url(&user, &password, TEST_BASE_URL)
            .expect("Failed to create client");

        // Use comparison operator: greater than
        // The ">" prefix means "greater than"
        let recent_records = client
            .gp(
                None,
                None,
                None,
                Some(">2024-01-01"), // epoch > 2024-01-01
                None,
                None,
                Some(5),
                None,
            )
            .expect("Query failed");

        println!("Found {} records with epoch > 2024-01-01", recent_records.len());

        // Use comparison operator: less than
        // The "<" prefix means "less than"
        let old_records = client
            .gp(
                None,
                None,
                None,
                Some("<2020-01-01"), // epoch < 2020-01-01
                None,
                None,
                Some(5),
                None,
            )
            .expect("Query failed");

        println!("Found {} records with epoch < 2020-01-01", old_records.len());
        // Found 5 records with epoch > 2024-01-01
        // Found 5 records with epoch < 2020-01-01
    }

    println!("Example completed successfully!");
}
